File: code_eval/java/easy/Challenge20.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 20 : Strings to Lower Case
 * Created by cschalk on 3/21/16.
 */
public class Challenge20 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            System.out.println(line.toLowerCase());
        }
    }
}

File: code_eval/cpp/easy/challenge_20.cpp
// Created by cschalk on 3/20/16.
// Challenge 20 : Strings to lower.
#include <fstream>
#include <iostream>
#include <locale>

using namespace std;

int main(int argc, char ** args) {
    std::ifstream input {args[1]};
    std::locale loc;
    std::string line;

    while (std::getline(input, line)) {
        for (std::string::size_type i = 0 ; i < line.length() ; ++i ) {
            std::cout << std::tolower(line[i], loc);
        }
        std::cout << std::endl;
    }

    return 0;

}
